Corvallis-Sustainability-Coalition/ReUse-Websites
# Directory structure update and scaffold view rendering

Changes Summary

- changed directory structure and folder names to saner
  defaults
- create simple working view renderer
- modify function for getting the database to use a static
  db to reduce the number of connections and increase speed
- update tests and travis script to reflect and use new directory structure

Testing Instructions:

- You should probably run a `composer update` and/or `composer dump-autoload` at the root of the project.  Also, if you get permissions issues with the xml files in `public_html/xml`, I've found running `chmod -R 777 public_html/xml` seems to fix that.  
- Running tests is now done with `vendor/bin/phpunit tests` and if you want more information throw the `--verbose` flag in ther.
- Finally, just give it a whirl and make sure I didn't miss anything major (_everything is loading correctly, the site still seems to work, the API isn't majorly broken, etc...).  

Directory structure is as follows:

```
{project root}
	bin - all executables / shell scripts
	config - currently empty but will / should house any and all configuration
	      files used by any part of the application
 	public_html - the apche document root.  Once proper view rendering is done,
	      this should hold publicly accessible assets (css, js, images, xml files)
	resources - non public assets used by the application
	src - all PHP files
		lib - all our own library code (non vendor extras)
		routes - all the routes used by Slim
		view - any / all files related to rendering the view
	tests - holds all implemented tests
	vendor - the composer generated vendor folder
```

Please note the following update to the how database access is handled:

	- All connections to the database are now handled through the same static
	  mysqli instance, so any time you need to run a query, just get the static
	  mysqli instance with a call to connectReuseDB(), and run your queries
	  **without** calling the close() method.  That leaves the connection open
	  for the next place it gets called.

View Rendering:

	- Presently, the views are still handled by the static PHP files and the
	  AJAX stuff in the JS.  Once we begin this transition, all the databse
	  access will happen in the PHP and the Slim framework will take care of
	  passing whatever gets returned from the databse to the view renderer to
	  be injected into the templates that the view calls.  For more details
	  on how the view renderer should work, see the slim documentation at:
	  http://docs.slimframework.com/view/overview/
